python/strings_udf/strings_udf/__init__.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+from ptxcompiler.patch import patch_needed, CMD
+import os
+import warnings
+import sys
+import subprocess
+
+
+def versions_compatible(path):
+    """
+    Example PTX header:
+
+    // Generated by NVIDIA NVVM Compiler
+    //
+    // Compiler Build ID: CL-30672275
+    // Cuda compilation tools, release 11.5, V11.5.119
+    // Based on NVVM 7
+    """
+    # obtain the cuda version used to compile this PTX file
+    file = open(path).read()
+    major, minor = file.split("\n")[4].split(" ")[-2][:-1].split(".")
+
+    # adapted from patch_needed()
+    cp = subprocess.run([sys.executable, "-c", CMD], capture_output=True)
```

Comment:
I wish there was a way to avoid creating a subprocess on import. Could we make `ENABLED` a function that caches the value and uses it only when needed?

Reply:
This is actually worse than it looks and I think there's a clean solution but I'm just not sure where to do it. Right now, a subprocess is opened actually three times by my eye:
- When cuDF is imported, [ptxcompiler does it to see if numba need be patched](https://github.com/rapidsai/cudf/blob/fb5af4be82146ea0c8a86b8780540610502c7076/python/cudf/cudf/__init__.py#L100)
- When `strings_udf` is imported, once from `patch_needed()` and once again to obtain the major and minor cuda versions to do our additional checking against the version of cuda required to compile the PTX file.

It seems like we're going to need to do it at least once, but shouldn't need to do it more than that. I could see us changing `ptxcompiler` perhaps to cache things like the cuda versions and make them available for external libraries to consume and do what they will with - but I can't tell if that's the right place for them all.

Reply:
I agree with your assessment but perhaps we should defer until later on this.